app/main/posts/views/filters/filter-saved-search.directive.js

```diff
@@ -17,28 +17,34 @@ function FilterSavedSearch(SavedSearchEndpoint, _,  $rootScope, ModalService) {
         scope.searchesLength = 0;
         scope.loading = false;
         scope.$on('savedSearch:update', loadSavedSearches);
-
+        scope.$watch(PostFilters.getModeId, function (newValue, oldValue) {
+            if (typeof (newValue) === 'undefined') {
+                scope.selectedSavedSearch = null;
+            } else if (oldValue !== newValue && scope.searches.length > 0) {
+                scope.selectedSavedSearch =  scope.searches[newValue];
+            }
+        });
         scope.openSavedSearchListEditorModal = function () {
             ModalService.openTemplate('<saved-search-list-editor-modal searches="searches"></saved-search-list-editor-modal>', 'set.delete_saved_searches', 'star', scope, false, false);
         };
 
         function activate() {
-            ngModel.$render = renderModelValue;
-            if (ngModel.$viewValue.length > 0) {
+            if (ngModel.$viewValue) {
                 scope.selectedSavedSearch = scope.searches[ngModel.$viewValue];
+            } else if (PostFilters.getModeId()) {
+                scope.selectedSavedSearch = scope.searches[PostFilters.getModeId()];
```

Comment:
this here attempts to solve the first load issue on saved searches. basically you should be able to get the modeid AFTER loading the .searches array which is what happens here in activate